compiler/rustc_codegen_llvm/src/common.rs

```diff
@@ -249,18 +261,38 @@ impl<'ll, 'tcx> ConstMethods<'tcx> for CodegenCx<'ll, 'tcx> {
                 let (base_addr, base_addr_space) = match self.tcx.global_alloc(alloc_id) {
                     GlobalAlloc::Memory(alloc) => {
                         let init = const_alloc_to_llvm(self, alloc);
+                        let debug_hint = alloc.1;
                         let alloc = alloc.inner();
                         let value = match alloc.mutability {
                             Mutability::Mut => self.static_addr_of_mut(init, alloc.align, None),
-                            _ => self.static_addr_of(init, alloc.align, None),
+                            _ => {
+                                let value = self.static_addr_of(init, alloc.align, None);
+                                llvm::set_linkage(value, llvm::Linkage::WeakODRLinkage);
+                                value
+                            }
                         };
-                        if !self.sess().fewer_names() && llvm::get_value_name(value).is_empty() {
+
+                        if llvm::get_value_name(value).is_empty() {
+                            let name_prefix = match debug_hint {
+                                Some(ConstAllocationDebugHint::StrLiteral) => "str_",
+                                Some(ConstAllocationDebugHint::CallerLocation) => "callerloc_",
+                                Some(ConstAllocationDebugHint::TypeName) => "typename_",
+                                Some(ConstAllocationDebugHint::VTable) => "vtable_",
+                                None => "alloc_",
```

Comment:
Please use a prefix which is clearly unique to rustc (and preferably to a single version of rustc in case staticlibs of different rustc versions are mixed)

Reply:
Done: feel free to provide suggestions on the new naming scheme